guidance/models/_azure_openai.py

```diff
@@ -1,67 +1,56 @@
-import re
-from urllib.parse import urlparse, parse_qs
+from urllib.parse import parse_qs, urlparse
 
+from ._grammarless import Grammarless
+from ._model import Chat, Instruct
 from ._openai import (
-    OpenAIChat,
-    OpenAICompletion,
-    OpenAIInstruct,
-    chat_model_pattern,
+    OpenAIChatEngine,
+    OpenAICompletionEngine,
+    OpenAIInstructEngine,
 )
-from ._openai import OpenAI
 
 try:
     # TODO: can we eliminate the torch requirement for llama.cpp by using numpy in the caller instead?
     import openai as openai_package
-    import tiktoken
 
     is_openai = True
 except ImportError:
     is_openai = False
 
 
-class AzureOpenAI(OpenAI):
+class AzureOpenAI(Grammarless):
     def __init__(
         self,
         model: str,
         azure_endpoint: str,
-        azure_deployment: str,
+        azure_deployment: str = None,
         azure_ad_token_provider=None,
         api_key: str = None,
         tokenizer=None,
```

Comment:
@riedgar-ms I assume there's no way to do auto-tokenizer detection logic on Azure OpenAI models?

Reply:
It will - it ultimately uses the `model` string, just like with the OpenAI class. Users to have to set this for themselves, though, since it's not part of the URI which AzureAI provides.